projects/kit/components/range/range.component.ts

```diff
@@ -167,10 +173,31 @@ export class TuiRange extends TuiControl<[number, number]> implements OnChanges
     }
 
     private updateStart(value: number): void {
-        this.onChange([Math.min(value, this.value()[1]), this.value()[1]]);
+        const newValue = Math.min(value, this.value()[1]);
+        const distance = this.value()[1] - newValue;
+
+        if (this.isDistanceExceeded(distance)) {
+            return;
+        }
+
+        this.onChange([newValue, this.value()[1]]);
     }
 
     private updateEnd(value: number): void {
-        this.onChange([this.value()[0], Math.max(value, this.value()[0])]);
+        const newValue = Math.max(value, this.value()[0]);
+        const distance = newValue - this.value()[0];
+
+        if (this.isDistanceExceeded(distance)) {
```

Comment:
I think `checkDistance` is a better name, because it checks both limits.

Reply:
Fixed